// babel-plugin-transform-fuse-requires/lib/index.js
"use strict";

var _interopRequireWildcard = require("babel-runtime/helpers/interop-require-wildcard")["default"];
var _babelTypes = require("babel-types");
var t = _interopRequireWildcard(_babelTypes);
var each = require("lodash/collection/each");
var ids = require("../../../lib/ids");

exports.__esModule = true;

exports["default"] = function () {
  return {
    visitor: {
      LVal: function LVal (path, file) {
        if (t.isReferenced(path.node, path.parent)) {
          console.log("t.isReferenced");
        }
        if (path.node.name === "global") {
          console.log("We have global");
          my_global = 1;

        }
      },
      Program: { 
        enter: function ProgramEnter (path, file) {
          my_global = 0;

        },
      exit: function ProgramExit (path, file) {
        console.log("Program exit");
        if (my_global) {
          console.log("We need to add global!!");
          // console.log(t.BUILDER_KEYS);
          var topNodes = [];
          // topNodes.push(t.variableDeclaration("var", [t.variableDeclarator(ref, buildRequire(t.stringLiteral()).expression)]));
          // topNodes.push(t.variableDeclaration("var", [t.variableDeclarator(t.stringLiteral()).expression]));
          topNodes.push(t.expressionStatement(t.stringLiteral("Is this the real life?")));
          //   AssignmentExpression: [ 'operator', 'left', 'right' ],
          // topNodes.push(t.assignmentExpression("=", t.variableDeclaration("var", [t.variableDeclarator(t.identifier("global"))])));
          topNodes.push(t.variableDeclaration("var", [t.variableDeclarator(t.identifier("global"))]));
          // push var global = this;
          path.unshiftContainer("body", topNodes);
        }
      } },
      CallExpression: function CallExpression (path, file) {
        if (path.node.callee.name === "require") {
          var req = path.node.arguments[0].value;
          var rewrite = ids.get_require(req);
          path.node.arguments[0].value = rewrite;
        }
      }
    }
  };
};

var my_global = 0;

module.exports = exports["default"];
// console.log(t.TYPES);
